geanyprj/src/project.c

```diff
@@ -463,5 +478,6 @@ void geany_project_save(struct GeanyPrj *prj)
 		g_hash_table_foreach(prj->tags, geany_project_save_files, &data);
 	}
 	save_config(config, prj->path);
+	set_non_ft_build_wdir(g_current_project->working_dir);
```

Comment:
When creating a new project based upon an open file, this is causing a segfault during my tests

```
Thread 1 "geany" received signal SIGSEGV, Segmentation fault.
0x00007fffe579e791 in geany_project_save (prj=prj@entry=0x555556805350) at project.c:481
481		set_non_ft_build_wdir(g_current_project->working_dir);
(gdb) bt
#0  0x00007fffe579e791 in geany_project_save (prj=prj@entry=0x555556805350) at project.c:481
#1  0x00007fffe579d4dc in on_new_project (menuitem=<optimized out>, user_data=<optimized out>) at menu.c:250
#2  0x00007ffff5995f75 in g_closure_invoke () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#3  0x00007ffff59a7f82 in ?? () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#4  0x00007ffff59b0bcc in g_signal_emit_valist () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#5  0x00007ffff59b0faf in g_signal_emit () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#6  0x00007ffff73a28de in gtk_widget_activate () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#7  0x00007ffff7270826 in gtk_menu_shell_activate_item () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#8  0x00007ffff7270b5b in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#9  0x00007ffff7253b11 in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#10 0x00007ffff59961a4 in ?? () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#11 0x00007ffff59b0391 in g_signal_emit_valist () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#12 0x00007ffff59b0faf in g_signal_emit () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#13 0x00007ffff73a03dc in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#14 0x00007ffff7250ace in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#15 0x00007ffff7252b5e in gtk_main_do_event () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#16 0x00007ffff6d63bf5 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#17 0x00007ffff6d94be2 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#18 0x00007ffff52b67f7 in g_main_context_dispatch () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#19 0x00007ffff52b6a60 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#20 0x00007ffff52b6d82 in g_main_loop_run () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#21 0x00007ffff7251cc5 in gtk_main () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#22 0x00007ffff79ad097 in main_lib (argc=<optimized out>, argv=<optimized out>) at libmain.c:1233
#23 0x00007ffff4cd12b1 in __libc_start_main () from /lib/x86_64-linux-gnu/libc.so.6
#24 0x000055555555478a in _start ()

```